frontend/lib/account-settings/nyc-address-settings.tsx

```diff
@@ -74,13 +83,42 @@ const PublicAssistanceField: React.FC<WithAccountSettingsProps> = ({
   );
 };
 
+function filterLeaseChoices(
+  leaseType: "" | LeaseType,
+  choices: [LeaseChoice, string][]
+) {
+  let filteredChoices = [...choices];
+
+  // If the user previously onboarded and chose the "No lease" or "Rent stabilized
+  // or controlled" options, which have since been deprecated, we still want them
+  // to be able to see it and choose it. However, if they didn't pick that option,
+  // we don't want them to be able to choose it now.
+  if (leaseType !== "NO_LEASE") {
+    filteredChoices = choices.filter(([choice, _]) => choice !== "NO_LEASE");
+  }
+  if (leaseType !== "RENT_STABILIZED_OR_CONTROLLED") {
+    filteredChoices = choices.filter(
+      ([choice, _]) => choice !== "RENT_STABILIZED_OR_CONTROLLED"
+    );
```

Comment:
I guess if we do much more of this kind of thing, maybe we should factor out an `onlyIncludeChoiceIfCurrentlySet()` function or somesuch?  Doesn't seem necessary for this PR though!

Reply:
I started down a rabbit hole of doing that and then i was like that's even more confusing. If this happens a lot can revisit it for sure tho.